libs/game/info.ts

```diff
@@ -766,7 +767,14 @@ namespace info {
 
         onScore(score: number, handler: () => void) {
             const state = this.getState();
-            state.scoreReachedHandler = new ScoreReachedHandler(score, handler);
+            state.scoreReachedHandler?.forEach(element => {
+                if (element.score === score) {
+                    // Score handlers are implemented as "last one wins."
+                    element.handler = handler;
+                    return;
```

Comment:
I don't think this will work as you intend, this will return in the anonymous fn for that value but still iteratw through the rest (and, more importantly, still push onto the array after the foreach is done)

Try a `for (const element of state.scoreReachedHandler){` instead of `[].forEach`and same code should work correctly

Reply:
thanks for the catch!